backend/app/platforms/router.py

```diff
@@ -0,0 +1,60 @@
+from typing import Sequence
+from uuid import UUID
+
+from fastapi import APIRouter, Depends, HTTPException, status
+from sqlalchemy.orm import Session
+
+from app.database.database import get_db_session
+from app.dependencies import only_for_admin
+
+from .schemas import (
+    GetPlatformServicesSchema,
+    GetPlatformsSchema,
+    PlatformServiceConfigSchema,
+)
+from .service import PlatformsService
+
+router = APIRouter(prefix="/platforms", tags=["platforms"])
+
+
+@router.get("", dependencies=[Depends(only_for_admin)])
+def get_all_platforms(
+    db: Session = Depends(get_db_session),
+) -> Sequence[GetPlatformsSchema]:
+    return PlatformsService(db).get_all_platforms()
+
+
+@router.get("/{platform_id}/services", dependencies=[Depends(only_for_admin)])
+def get_platform_services(
+    platform_id: UUID, db: Session = Depends(get_db_session)
+) -> Sequence[GetPlatformServicesSchema]:
+    return PlatformsService(db).get_platform_services(platform_id)
+
+
+@router.get(
+    "/{platform_id}/services/{service_id}",
+    dependencies=[Depends(only_for_admin)],
+    responses={
+        404: {
+            "description": "Service configuration not found",
+            "content": {
+                "application/json": {
+                    "example": {"detail": "Service configuration not found"}
+                }
+            },
+        }
+    },
+)
+def get_service_config(
+    platform_id: UUID, service_id: UUID, db: Session = Depends(get_db_session)
+) -> PlatformServiceConfigSchema:
+    if not (
```

Comment:
this method should not contain the logic. The 404 functionality should happen inside of the service

Reply:
I can't entirely agree. I think all fastapi-related logic should be part of a router and not of a service. What if I need to reuse the service in another place and I want to check that a record doesn't exist? Why should the service return HTTPException?

Reply:
This should never be the case right? The 404 happens because something unexpected occurs. It should not be thrown in case we deliberately check if something exists. Then you should use another method. The get_service_config always expects the service config to be there, so it raises an exception if it is not.